### ISSUE TITLE
Current `main` branch introduces stutter

### ISSUE THREAD

**Issue by CleanCut** [Toqozz/bevy_debug_lines#14]
I'm experiencing stutter on my best hardware with the current `main` branch at 7859f9b87583dc175e43f9a54af50d340f92f7a9. I was able to resolve the stutter by removing `DebugLinesPlugin`.  

```diff
diff --git a/src/game.rs b/src/game.rs
index a910174..2cf7c20 100644
--- a/src/game.rs
+++ b/src/game.rs
@@ -224,7 +224,7 @@ pub fn update_window_dimensions(windows: Res<Windows>, mut engine_state: ResMut<
 #[doc(hidden)]
 pub fn draw_sprite_colliders(
     engine_state: Res<EngineState>,
-    mut lines: ResMut<DebugLines>,
+    //mut lines: ResMut<DebugLines>,
     sprite_query: Query<&Sprite>,
 ) {
     if !engine_state.debug_sprite_colliders {
@@ -239,7 +239,7 @@ pub fn draw_sprite_colliders(
         let mut curr = 0;
         let mut next = 1;
         while curr < length {
-            lines.line(points[curr].extend(0.0), points[next].extend(0.0), 0.0);
+            //lines.line(points[curr].extend(0.0), points[next].extend(0.0), 0.0);
             curr += 1;
             next = (next + 1) % length;
         }
@@ -306,7 +306,7 @@ impl<S: Send + Sync + 'static> Game<S> {
             .add_system(exit_on_esc_system)
             // External Plugins
             .add_plugin(AudioPlugin) // kira_bevy_audio
-            .add_plugin(DebugLinesPlugin::always_in_front()) // bevy_prototype_debug_lines, for displaying sprite colliders
+            //.add_plugin(DebugLinesPlugin::always_in_front()) // bevy_prototype_debug_lines, for displaying sprite colliders
             // Rusty Engine Plugins
             .add_plugin(AudioManagerPlugin)
             .add_plugin(KeyboardPlugin)
```

The weirdest part was that the stutter is in an example program which isn't using debug lines (it is literally not calling `lines.line(...)` from the diff above) because I've got code at the top of the function returning early and skipping the debug line drawing entirely (see code below). So I've got stutter...and the plugin ostensibly isn't even _doing anything_!

```rust
   if !engine_state.debug_sprite_colliders {
        return;
    }
```
 
The stutter happens randomly with an interval that feels like it's random between about 0.25s - 5s. The stutter seems to last at least a couple frames and sometimes more like a dozen.

Here's the specs for my workstation with the problem:

<img width="245" alt="image" src="https://user-images.githubusercontent.com/5838512/149678581-8a447bf8-b441-4db1-ba62-fb9379a5e741.png">

```
# I tried Rust 1.57 -- same thing
$ rustc --version
rustc 1.58.0 (02072b482 2022-01-11)
```

I do NOT see the problem on my laptop, which is really weird:

<img width="239" alt="mbpro-screenshot" src="https://user-images.githubusercontent.com/5838512/149679193-d9337d74-6ba4-4b8e-ba49-ec8b16f9f0f3.png">

# Steps to Reproduce

```
$ git clone https://github.com/CleanCut/rusty_engine.git
$ cd rusty_engine
$ git checkout choppy
$ cargo run --release --example road_race

# Use the arrow keys to avoid obstacles - observe periodic stutters on some hardware

$ git checkout smooth
$ cargo run --release --example road_race

# no stutters now - neither of these will draw any debug lines (`engine_state.debug_sprite_colliders`
  is `false` in this example)
```

**Comment by Toqozz** [comment]
Hey!  Thanks for the issue.

This is a weird issue, but I'm not too surprised by it.  The upgrade to `bevy 0.6` introduced huge changes (#10) and there are definitely still some kinks to work out.  I'm really busy lately but am going to be working through them.

Thanks for the detailed report!

**Comment by CleanCut** [comment]
More info:

- I tried ripping out `bevy_prototype_debug_lines` and using `bevy_prototype_lyon` instead...and the _exact_ same stutter behavior starts up again.  AFAIK the thing in common between these two plugins is that they use custom shaders. Given the massive rendering overhaul in Bevy `0.6`, this may be more of a Bevy issue than anything. 🤔 It would still be great if we could _find_ that issue!
- I patched `Cargo.toml` to use the current Bevy `main` branch (currently on commit [3fcdc5a](https://github.com/bevyengine/bevy/commit/3fcdc5a4926dcad89a3346977d862c505993d265)), and the problem persists...so nothing merged into `main` since the `0.6.0` release has fixed it, yet.

**Comment by CleanCut** [comment]
I filed [an issue on `bevyengine/bevy`](https://github.com/bevyengine/bevy/issues/3709) as well, since it could very well be an upstream problem.

**Comment by CleanCut** [comment]
The issue is most certainly upstream Bevy 0.6. Feel free to close this, or leave it open until upstream is resolved. Whatever you prefer.

**Comment by Toqozz** [comment]
Leaving it open :)

**Comment by adsick** [comment]
Hi, for me it seems like stutters are present from the very beginning (in my case from versions 0.3-0.4). I've observed noticeable drops with 2-5 seconds interval since those version and thought "well, some day this will be fixed"

hardware: amd ryzen 5 3500U, Vega 8 graphics, Fedora Linux 35. (also I remember it on Windows with bevy 0.5)

**Comment by Toqozz** [comment]
I've never noticed any stutter on my end.  The way the plugin works (especially prior to 0.6.0) is super simple, so I don't see many places that could cause a stutter.  Maybe a memory bandwidth issue due to the integrated graphics?  But it seems unlikely.

We split lines between multiple meshes in 0.6.0, which might help with that.  Would you mind trying it out?

**Comment by adsick** [comment]
> We split lines between multiple meshes in 0.6.0, which might help with that.

what do you mean?

**Comment by CleanCut** [comment]
Resolved in Bevy 0.8
